students/arun_nalla/session03/Lesson3_Mailroom.py

```diff
@@ -1,62 +1,58 @@
-#!usr/bin/en python
-donors_list = [('Aron 123', [10,5]), ('Becky', [3,30]), ('Charlie',[10,26]),('Thomas',[22,9,4,5]),('Nancy',[1,2,3])]
-donors_name = [x for x,y in donors_list]
-donors_number = {x: len(y) for x,y in donors_list}
-donors_total = {x:sum(y) for x,y in donors_list}
-
-def thank_you(seq):
-    name_input = input("Enter a name from the list: ").title()
-    if name_input in donors_name:
-        print("\nDear Mr.{},\nThank you for donation of ${:.2f} for XYZ Charity Organisation. Its a kind initiative from"
-        " you for a nobel cause.\n\nWith Regards\n\nXYZ\n".format(name_input, donors_total[name_input]))
-    else:
-        print ('Name entered is not in the donor list')
-        thank_you(seq)
-def donors_thanks(seq):
-    name_input = input("Enter the donor of your interest: ").title()
-    if name_input == 'List':
-        print (donors_name)
-        thank_you(seq)
-    elif name_input in donors_name:
-        print("Dear Mr.{},\nThank you for donation of ${:.2f} for XYZ Charity Organisation. Its a kind intiative from"
-        " you for a nobel cause.\n\nWith Regards\n\nXYZ\n".format(name_input, donors_total[name_input]))
-    elif name_input !=seq:
-        seq.append(name_input)
-        print("Dear {},\nI am writing regarding the charity program 'XYZ' for 'ABC' cause. \nIf you are interested"
-              " please reply back. Your help will be appreciated.\n\nThank you\nXYZ\n".format(name_input))
+#!usr/bin/env python
+donors_list = [('Aron King', [1,2]), ('Becky Patty',[3000,1000]), ('Charlie Lee',[666,222]),('Thomas Dave',[202,90,450]),('Nancy Crow',[100,200,300])]
 
+def thanks_email(seq):
+    for i, v in enumerate(seq):
+        while True:
```

Comment:
Hmm... I think these two loops are reversed. Actually, not just reversed, but the `for` loop shouldn't even apply if `name_input == 'List'`, don't you think?

The loop control here is totally flawed. You've got the `for` loop that iterates over all items in `seq`, but that works only when `name_input == 'List'`, which doesn't even apply when the user enters a correct/existing name. When the user enters a name that already exists in the database, your line 18 will get activated no matter what (assuming the entered name is not at index 0), which will make just append another entry in the list for the same name, and that's not what was asked.

Note that the `for` loop is to search the `seq` for the entered `name_input`, when it's not `'List'`. So you'll need to rewrite the loops correctly to get this fixed.